### PR TITLE
Add filter to force decimal points in refund form

### LINKED ISSUE #2577: Wrong display of decimal points in order refund form
When you try to refund an order with MC enabled and having a different customer currency, the refund input content doesn't match with the price above, it doesn't seem to respect the currency's decimal places count.

ezgif com-gif-maker

The source of it looks like https://github.com/woocommerce/woocommerce/blob/f8955593f16b446f99bc14170e4f0ad51fb63700/includes/admin/meta-boxes/views/html-order-fee.php#L46-L48

Which gets the price attribute from https://github.com/woocommerce/woocommerce/blob/f8955593f16b446f99bc14170e4f0ad51fb63700/assets/js/admin/meta-boxes-order.js#L1031-L1035

Which is rendered by https://github.com/woocommerce/woocommerce/blob/2cce7f3af89760dc89d96844c748ae646a658ceb/includes/wc-formatting-functions.php#L343-L345

So if we can hook into `wc_format_localized_price` to adjust the decimal points, I think it'll be fixed.

### PR DESCRIPTION
**Pull request by tpaksu**
Fixes #2577

#### Changes proposed in this Pull Request

The decimal places in a price wasn't taken into account <strike>when formatting the price with `wc_format_localized_price`. This adds a filter to the `woocommerce_format_localized_price` hook to ensure that the decimal points of the output matches the store currency.</strike>

The number of decimal places are defined by `WC_ROUNDING_PRECISION` constant when calculating the input value in the refund form here: https://github.com/woocommerce/woocommerce/blob/32993b8aaf36be0c03c5a572e83ee61bb90927e6/assets/js/admin/meta-boxes-order.js#L372-L384

This PR overrides that constant by appending a script to the `<head>` element when MC has _more than one currency available for use_ and the current page is the **WooCommerce > Orders > Order Details** page:

```php
/**
	 * Reduces the client rounding precision to 2
	 *
	 * @return  void
	 */
	public function set_client_rounding_precision() {
		return;
		$screen = get_current_screen();
		if ( 'post' === $screen->base && 'shop_order' === $screen->post_type ) :
			$rounding_precision = wc_get_price_decimals() ?? wc_get_rounding_precision();
			?>
		<script>woocommerce_admin_meta_boxes.rounding_precision = <?php echo intval( $rounding_precision ); ?>;</script>
			<?php
		endif;
	}
``` 

### Before the fix is applied:
before_fix

### After the fix is applied:
after_fix

#### Testing instructions

* Go to any order with an item having a price with fractions
* Try to refund the item, on the number input, increase the refund product number to 1
* Check if the price input contains a value with the same decimal places like the price shown above of it.

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by tpaksu** [comment]
Finding out that it doesn't work as expected and JS side has the full control of it, PHP filter doesn't have the chance to intervene this output, as we talked in this thread (p1627643718019900-slack-C01B8KNUYSW) this issue will need further testing so it's better to leave it into Phase 2.

**Comment by tpaksu** [comment]
After digging some more, I could find a way to override the `woocommerce_admin_meta_boxes.rounding_precision` variable by hooking into the `<head>` output and appending a script that overrides the value of it to the admin currency decimal places. Added the fix, asked @claudiosanches about the importance of the `WC_ROUNDING_PRECISION` constant, which seems to be only used in that form's client code, waiting for an answer. I would prefer having an hook to `wcs_get_rounding_precision` function to override it, but it seems there are none affecting the output of that function.

**Review by dmallory42 — APPROVED** [review]
LGTM and works well. I left one comment for consideration but happy to 🚢  depending on what you think. 👍

File: includes/multi-currency/MultiCurrency.php [hunk 1]
```diff
@@ -441,6 +441,10 @@ private function initialize_available_currencies() {
 		foreach ( $available_currencies as $currency ) {
 			$this->available_currencies[ $currency->get_code() ] = $currency;
 		}
+
+		if ( is_admin() && 1 < count( $this->available_currencies ) ) {
+			add_action( 'admin_head', [ $this, 'set_client_rounding_precision' ] );
```
[the remark this answers is not in the document]
Comment (tpaksu) [reply includes/multi-currency/MultiCurrency.php]: I think you tested an outdated version of this, as I couldn't find this check :) Can you please check if you have the latest one? 1ed4f9b?

**Pull request merged**